lib/class-wp-rest-block-renderer-controller.php

```
@@ -0,0 +1,145 @@
+							'description'          => sprintf( __( 'Attributes for %s block', 'gutenberg' ), $block_type->name ),
+							'type'                 => 'object',
+							'additionalProperties' => false,
+							'properties'           => $block_type->attributes,
+						),
+					),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			) );
+		}
+	}
+
+	/**
+	 * Checks if a given request has access to read blocks.
+	 *
+	 * @since ?
+	 * @access public
+	 *
+	 * @param WP_REST_Request $request Request.
+	 * @return true|WP_Error True if the request has read access, WP_Error object otherwise.
+	 */
+	public function get_item_permissions_check( $request ) {
+		global $post;
+
+		$post_ID = isset( $request['post_id'] ) ? intval( $request['post_id'] ) : 0;
+
+		if ( 0 < $post_ID ) {
+			$post = get_post( $post_ID );
+			if ( ! $post || ! current_user_can( 'edit_post', $post->ID ) ) {
+				return new WP_Error( 'gutenberg_block_cannot_read', __( 'Sorry, you are not allowed to read Gutenberg blocks of this post', 'gutenberg' ), array(
+					'status' => rest_authorization_required_code(),
+				) );
+			}
+
+			// Set up postdata since this will be needed if post_ID was set.
+			setup_postdata( $post );
+
+		} else {
+			if ( ! current_user_can( 'edit_posts' ) ) {
+				return new WP_Error( 'gutenberg_block_cannot_read', __( 'Sorry, you are not allowed to read Gutenberg blocks as this user.', 'gutenberg' ), array(
+					'status' => rest_authorization_required_code(),
+				) );
+			}
+		}
+
+		return true;
+	}
+
+	/**
+	 * Returns block output from block's registered render_callback.
+	 *
+	 * @since ?
+	 * @access public
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function get_item( $request ) {
+		$registry = WP_Block_Type_Registry::get_instance();
+		$block    = $registry->get_registered( $request['name'] );
```

Can you return an error here if the block is invalid?